pkg/cmd/pr/checks/output.go

```diff
@@ -0,0 +1,108 @@
+package checks
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/cli/cli/v2/utils"
+)
+
+func addRow(tp utils.TablePrinter, isTerminal bool, cs *iostreams.ColorScheme, o check) {
+	elapsed := ""
+	zeroTime := time.Time{}
+
+	if o.StartedAt != zeroTime && o.CompletedAt != zeroTime {
+		e := o.CompletedAt.Sub(o.StartedAt)
+		if e > 0 {
+			elapsed = e.String()
+		}
+	}
+
+	mark := "✓"
+	markColor := cs.Green
+	switch o.Bucket {
+	case "fail":
+		mark = "X"
+		markColor = cs.Red
+	case "pending":
+		mark = "*"
+		markColor = cs.Yellow
+	case "skipping":
+		mark = "-"
+		markColor = cs.Gray
+	}
+
+	if isTerminal {
+		tp.AddField(mark, nil, markColor)
+		tp.AddField(o.Name, nil, nil)
+		tp.AddField(elapsed, nil, nil)
+		tp.AddField(o.Link, nil, nil)
+	} else {
+		tp.AddField(o.Name, nil, nil)
+		tp.AddField(o.Bucket, nil, nil)
+		if elapsed == "" {
+			tp.AddField("0", nil, nil)
+		} else {
+			tp.AddField(elapsed, nil, nil)
+		}
+		tp.AddField(o.Link, nil, nil)
+	}
+
+	tp.EndRow()
+}
+
+func printTable(isTerminal bool, cs *iostreams.ColorScheme, meta checksMeta, checks []check, opts *ChecksOptions) error {
```

Comment:
Similar to the `collect` function this feels like it is doing too much work. The output of the summary and table are two distinct parts of this function so that seems like a good place to break it up. What do you think?